pkg/client/apiutil/filterdiscoveryclient.go

```diff
@@ -0,0 +1,104 @@
+package apiutil
+
+import (
+	openapi_v2 "github.com/googleapis/gnostic/OpenAPIv2"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+)
+
+// defaultRetries is the number of times a resource discovery is repeated if an api group disappears on the fly (e.g. ThirdPartyResources).
+const defaultRetries = 2
+
+type FilterDiscoveryClient struct {
+	Discovery discovery.DiscoveryInterface
+	Filter    func(groupVersion string) bool
```

Comment:
especially this: does `true` mean `keep`, or `throw away`?

Reply:
something something boolean blindness something something

Reply:
Added a type FilterGroupFunc with godoc that explains the api.